spec/lib/audit/checksum_spec.rb

```diff
@@ -1,14 +1,48 @@
 require 'rails_helper'
 require_relative "../../../lib/audit/checksum.rb"
+require_relative '../../load_fixtures_helper.rb'
+
 # FIXME: remove this rubocop once we start writing tests
 # TODO: implement this;  we begin with a placeholder
 
 RSpec.describe Checksum do
-  let(:endpoint_name) { 'fixture_sr3' }
+  let(:endpoint_name) { 'fixture_sr1' }
+  let(:limit) { Settings.c2m_sql_limit }
+
+  context '.validate_disk' do
+    include_context 'fixture moabs in db'
+    let(:subject) { described_class.validate_disk(endpoint_name, limit) }
+
+    context 'when there are PreservedCopies to check' do
+      let(:cv_mock) { instance_double(ChecksumValidator) }
+
+      it 'creates an instance and calls #validate_checksum for every result when results are in a single batch' do
+        allow(ChecksumValidator).to receive(:new).and_return(cv_mock)
+        expect(cv_mock).to receive(:validate_checksum).exactly(3).times
+        described_class.validate_disk(endpoint_name, limit)
+      end
 
-  describe ".validate_disk" do
-    described_class.validate_disk('fixture_sr3')
-    skip 'we should figure out what they are and test them'
+      it 'creates an instance and calls #validate_checksum on everything in batches' do
```

Comment:
should we check the ordering here, since changing it to .for_each would mess up the ordering?

Reply:
@jmartin-sul maybe you can help me answer this. I used the same tests as C2m_class_spec.rb... and it doesn't look like c2m is checking order either?

Reply:
maybe there should be such a test in both places to ensure that future refactoring doesn't mess it up ...

Reply:
so in the refactoring i'm doing where the `limit`/`each` construct gets factored out into a common method, i add some tests for that method, one of which checks ordering.  so maybe that will suffice, since i'll rebase on this PR and refactor a bit to pull out the `limit`/`each` that's being added inline for this work?

i'd agree that we should test ordering, though.  and indeed, it seems we missed testing that in the C2M specs (though we do make sure that all expected records are processed).

Reply:
see https://github.com/sul-dlss/preservation_catalog/pull/598/files#diff-a4c754f158c0a1ba32651fb058bcf56fR44

Reply:
Should I add a test for ordering, or do we think John's test that checks the ordering will suffice?

Reply:
i'd vote for letting the PR i linked take care of it, and trusting that the tests for the commonly used method will be sufficient, but i'm open to testing ordering in both of the clients too, if e.g. @ndushay has a strong preference for that.